internal/backup.go

```diff
@@ -400,27 +401,39 @@ func GetBackupsAndGarbageWithTarget(folder storage.Folder, targetPath string) (b
 		return nil, nil, err
 	}
 
-	sortTimes := GetBackupTimeSlices(backupObjects)
+	sortTimes := GetBackupTimeSlices(backupObjects, folder)
 	garbage = getGarbageFromPrefix(subFolders, sortTimes)
 
 	return sortTimes, garbage, nil
 }
 
 // TODO : unit tests
-func GetBackupTimeSlices(backups []storage.Object) []BackupTime {
+func GetBackupTimeSlices(backups []storage.Object, folder storage.Folder) []BackupTime {
 	sortTimes := make([]BackupTime, len(backups))
+	sortByCreationTime := true
 	for i, object := range backups {
 		key := object.GetName()
 		if !strings.HasSuffix(key, utility.SentinelSuffix) {
 			continue
 		}
-		time := object.GetLastModified()
-		sortTimes[i] = BackupTime{utility.StripBackupName(key), time,
-			utility.StripWalFileName(key)}
+		metaData, err := GetBackupMetaData(folder, key, utility.BaseBackupPath)
+		var creationTime time.Time = time.Time{}
```

Comment:
Too many things are taking place in this function. Let's decompose them: 
- extract the creation time retrieval logic
- extract the backups sorting logic